### ISSUE TITLE
Build fail on Android, deprecated Gradle features

### ISSUE THREAD

**Issue by Datagrdn** [flyskywhy/react-native-gcanvas#2]
Hey there, including the build log below. Main error is:

`> Task :android:gcanvas_library:mergeDebugNativeLibs FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.`

Looks very similar to [this issue](https://github.com/facebook/react-native/issues/29829)

I'm working on solving it, including the details here in case you can see it as something specific to this project.

`* What went wrong:
Execution failed for task ':android:gcanvas_library:mergeDebugNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > More than one file was found with OS independent path 'lib/armeabi-v7a/libfreetype.so'. If you are using jniLibs and CMake IMPORTED targets, see https://developer.android.com/studio/preview/features#automatic_packaging_of_prebuilt_dependencies_used_by_cmake`

`➜  react git:(newCanvas) ✗ react-native run-android
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
(node:14395) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
Jetifier found 1481 file(s) to forward-jetify. Using 4 workers...
warn JS server not recognized, continuing with build...
* daemon not running; starting now at tcp:5037
* daemon started successfully
info Installing the app...
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :app
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed in version 5.0 of the Android Gradle plugin.
For more information, see http://d.android.com/r/tools/update-dependency-configurations.html.

> Configure project :android:gcanvas_library
publishNonDefault is deprecated and has no effect anymore. All variants are now published.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
WARNING: The specified Android SDK Build Tools version (28.0.3) is ignored, as it is below the minimum supported version (29.0.2) for Android Gradle Plugin 4.0.1.
Android SDK Build Tools 29.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '28.0.3'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.

> Task :android:gcanvas_library:compileDebugJavaWithJavac

> Task :android:adapters:gcanvas_imageloader_fresco:compileDebugJavaWithJavac

> Task :flyskywhy_react-native-gcanvas:compileDebugJavaWithJavac

> Task :react-native-community_async-storage:compileDebugJavaWithJavac

> Task :react-native-community_slider:compileDebugJavaWithJavac

> Task :react-native-view-pdf:compileDebugJavaWithJavac

> Task :react-native-webview:compileDebugJavaWithJavac

> Task :android:gcanvas_library:externalNativeBuildDebug
Build gcanvas_armeabi-v7a
ninja: Entering directory `/Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@flyskywhy/react-native-gcanvas/android/gcanvas_library/.cxx/cmake/debug/armeabi-v7a'
[1/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/png/PngLoader.cpp.o
[2/46] Building C object core/CMakeFiles/gcanvas.dir/android/png/thirdparty/lodepng.c.o
[3/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/png/LodePngCoder.cpp.o
[4/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/view/GcanvasWeexAndroid.cpp.o
[5/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/g3d_jni.cpp.o
[6/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/elf_sym_reader.cpp.o
../../../../../../core/android/elf_sym_reader.cpp:66:33: warning: if statement has empty body [-Wempty-body]
                        ABORT("error size (%d)",size);
                                                     ^
../../../../../../core/android/elf_sym_reader.cpp:66:33: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:70:32: warning: if statement has empty body [-Wempty-body]
                ABORT("error size (%d)",size);
                                             ^
../../../../../../core/android/elf_sym_reader.cpp:70:32: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:97:33: warning: if statement has empty body [-Wempty-body]
                        ABORT("error size (%d)",size);
                                                     ^
../../../../../../core/android/elf_sym_reader.cpp:97:33: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:102:32: warning: if statement has empty body [-Wempty-body]
                ABORT("error size (%d)",size);
                                             ^
../../../../../../core/android/elf_sym_reader.cpp:102:32: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:671:58: warning: if statement has empty body [-Wempty-body]
                ABORT("open /system/lib/libart.so or libaoc.so error!");
                                                                       ^
../../../../../../core/android/elf_sym_reader.cpp:671:58: note: put the semicolon on a separate line to silence this warning
5 warnings generated.
[7/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/gmanager.cpp.o
[8/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/GCanvasLinkNative.cpp.o
[9/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/view/grenderer.cpp.o
../../../../../../core/android/3d/view/grenderer.cpp:371:58: warning: '&&' within '||' [-Wlogical-op-parentheses]
         info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) &&
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../../../../../../core/android/3d/view/grenderer.cpp:371:58: note: place parentheses around the '&&' expression to silence this warning
         info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) &&
                                                         ^
1 warning generated.
[10/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/GCanvasJNI.cpp.o
../../../../../../core/android/GCanvasJNI.cpp:78:78: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
                                                            (char *) soPath, "libweexcore.so");
                                                                             ^
1 warning generated.
[11/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvas.cpp.o
[12/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/manager/GCanvasManagerAndroid.cpp.o
[13/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvasManager.cpp.o
[14/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GCanvasState.cpp.o
[15/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GFontStyle.cpp.o
[16/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GConvert.cpp.o
[17/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GFrameBufferObject.cpp.o
[18/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvasWeex.cpp.o
../../../../../../core/src/GCanvasWeex.cpp:1176:21: warning: expression result unused [-Wunused-value]
                    (FillStyleRadialGradient *) (mCanvasContext->mCurrentState->mFillStyle)->AddColorStop(
                    ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../core/src/GCanvasWeex.cpp:1217:21: warning: expression result unused [-Wunused-value]
                    (FillStyleLinearGradient *) (mCanvasContext->mCurrentState->mFillStyle)->AddColorStop(
                    ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../core/src/GCanvasWeex.cpp:1468:25: warning: use of logical '&&' with constant operand [-Wconstant-logical-operand]
    return (type >> 29) && 0x1;
                        ^  ~~~
../../../../../../core/src/GCanvasWeex.cpp:1468:25: note: use '&' for a bitwise operation
    return (type >> 29) && 0x1;
                        ^~
                        &
../../../../../../core/src/GCanvasWeex.cpp:1468:25: note: remove constant to silence this warning
    return (type >> 29) && 0x1;
                       ~^~~~~~
3 warnings generated.
[19/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GStrSeparator.cpp.o
[20/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GGlyphCache.cpp.o
[21/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GTreemap.cpp.o
[22/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GCanvas2dContext.cpp.o
[23/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GShader.cpp.o
[24/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GShaderManager.cpp.o
[25/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GTexture.cpp.o
[26/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GPath.cpp.o
[27/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvas2DContextImpl.cpp.o
[28/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvasAndroid.cpp.o
[29/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvas2DContextAndroid.cpp.o
[30/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GWebglContext.cpp.o
[31/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFont.cpp.o
[32/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontFamily.cpp.o
../../../../../../core/src/platform/Android/GFontFamily.cpp:183:16: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
        return "NotoSansCJK-Regular.ttc";
               ^
1 warning generated.
[33/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontCache.cpp.o
[34/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/CharacterSet.cpp.o
[35/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Encode.cpp.o
[36/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontManagerImpl.cpp.o
[37/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFrameBufferObjectImpl.cpp.o
[38/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontManagerAndroid.cpp.o
[39/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GPreCompiledShaders.cpp.o
[40/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/FileUtils.cpp.o
[41/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Util.cpp.o
[42/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/GLUtil.cpp.o
[43/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Log.cpp.o
[44/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GSystemFontInformation.cpp.o
[45/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Value.cpp.o
[46/46] Linking CXX shared library ../../../../build/intermediates/cmake/debug/obj/armeabi-v7a/libgcanvas.so
Build gcanvas_arm64-v8a
ninja: Entering directory `/Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@flyskywhy/react-native-gcanvas/android/gcanvas_library/.cxx/cmake/debug/arm64-v8a'
[1/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/png/PngLoader.cpp.o
[2/46] Building C object core/CMakeFiles/gcanvas.dir/android/png/thirdparty/lodepng.c.o
[3/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/png/LodePngCoder.cpp.o
[4/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/view/GcanvasWeexAndroid.cpp.o
[5/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/g3d_jni.cpp.o
[6/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/elf_sym_reader.cpp.o
../../../../../../core/android/elf_sym_reader.cpp:66:33: warning: if statement has empty body [-Wempty-body]
                        ABORT("error size (%d)",size);
                                                     ^
../../../../../../core/android/elf_sym_reader.cpp:66:33: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:70:32: warning: if statement has empty body [-Wempty-body]
                ABORT("error size (%d)",size);
                                             ^
../../../../../../core/android/elf_sym_reader.cpp:70:32: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:97:33: warning: if statement has empty body [-Wempty-body]
                        ABORT("error size (%d)",size);
                                                     ^
../../../../../../core/android/elf_sym_reader.cpp:97:33: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:102:32: warning: if statement has empty body [-Wempty-body]
                ABORT("error size (%d)",size);
                                             ^
../../../../../../core/android/elf_sym_reader.cpp:102:32: note: put the semicolon on a separate line to silence this warning
../../../../../../core/android/elf_sym_reader.cpp:671:58: warning: if statement has empty body [-Wempty-body]
                ABORT("open /system/lib/libart.so or libaoc.so error!");
                                                                       ^
../../../../../../core/android/elf_sym_reader.cpp:671:58: note: put the semicolon on a separate line to silence this warning
5 warnings generated.
[7/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/gmanager.cpp.o
[8/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/3d/view/grenderer.cpp.o
../../../../../../core/android/3d/view/grenderer.cpp:371:58: warning: '&&' within '||' [-Wlogical-op-parentheses]
         info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) &&
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
../../../../../../core/android/3d/view/grenderer.cpp:371:58: note: place parentheses around the '&&' expression to silence this warning
         info.format != ANDROID_BITMAP_FORMAT_RGBA_8888) &&
                                                         ^
1 warning generated.
[9/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/GCanvasLinkNative.cpp.o
[10/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/GCanvasJNI.cpp.o
../../../../../../core/android/GCanvasJNI.cpp:78:78: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
                                                            (char *) soPath, "libweexcore.so");
                                                                             ^
1 warning generated.
[11/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvas.cpp.o
[12/46] Building CXX object core/CMakeFiles/gcanvas.dir/android/manager/GCanvasManagerAndroid.cpp.o
[13/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvasManager.cpp.o
[14/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GCanvasState.cpp.o
[15/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GFontStyle.cpp.o
[16/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GConvert.cpp.o
[17/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/GCanvasWeex.cpp.o
../../../../../../core/src/GCanvasWeex.cpp:1176:21: warning: expression result unused [-Wunused-value]
                    (FillStyleRadialGradient *) (mCanvasContext->mCurrentState->mFillStyle)->AddColorStop(
                    ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../core/src/GCanvasWeex.cpp:1217:21: warning: expression result unused [-Wunused-value]
                    (FillStyleLinearGradient *) (mCanvasContext->mCurrentState->mFillStyle)->AddColorStop(
                    ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../core/src/GCanvasWeex.cpp:1468:25: warning: use of logical '&&' with constant operand [-Wconstant-logical-operand]
    return (type >> 29) && 0x1;
                        ^  ~~~
../../../../../../core/src/GCanvasWeex.cpp:1468:25: note: use '&' for a bitwise operation
    return (type >> 29) && 0x1;
                        ^~
                        &
../../../../../../core/src/GCanvasWeex.cpp:1468:25: note: remove constant to silence this warning
    return (type >> 29) && 0x1;
                       ~^~~~~~
3 warnings generated.
[18/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GFrameBufferObject.cpp.o
[19/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GStrSeparator.cpp.o
[20/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GGlyphCache.cpp.o
[21/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GTreemap.cpp.o
[22/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GCanvas2dContext.cpp.o
[23/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GShader.cpp.o
[24/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GShaderManager.cpp.o
[25/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GTexture.cpp.o
[26/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GPath.cpp.o
[27/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvas2DContextImpl.cpp.o
[28/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvasAndroid.cpp.o
[29/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/gcanvas/GWebglContext.cpp.o
[30/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GCanvas2DContextAndroid.cpp.o
[31/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFont.cpp.o
[32/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontFamily.cpp.o
../../../../../../core/src/platform/Android/GFontFamily.cpp:183:16: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
        return "NotoSansCJK-Regular.ttc";
               ^
1 warning generated.
[33/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontCache.cpp.o
[34/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/CharacterSet.cpp.o
[35/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Encode.cpp.o
[36/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontManagerImpl.cpp.o
[37/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFrameBufferObjectImpl.cpp.o
[38/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GFontManagerAndroid.cpp.o
[39/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GPreCompiledShaders.cpp.o
[40/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/FileUtils.cpp.o
[41/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/platform/Android/GSystemFontInformation.cpp.o
[42/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Util.cpp.o
[43/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/GLUtil.cpp.o
[44/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Log.cpp.o
[45/46] Building CXX object core/CMakeFiles/gcanvas.dir/src/support/Value.cpp.o
[46/46] Linking CXX shared library ../../../../build/intermediates/cmake/debug/obj/arm64-v8a/libgcanvas.so

> Task :android:gcanvas_library:mergeDebugNativeLibs FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.1.1/userguide/command_line_interface.html#sec:command_line_warnings
227 actionable tasks: 218 executed, 9 up-to-date
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@flyskywhy/react-native-gcanvas/android/adapters/gcanvas_imageloader_fresco/src/main/java/com/taobao/gcanvas/adapters/img/impl/fresco/GCanvasFrescoImageLoader.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@flyskywhy/react-native-gcanvas/android/src/main/java/com/taobao/gcanvas/bridges/rn/GReactViewManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@react-native-community/async-storage/android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/react-native-view-pdf/android/src/main/java/com/rumax/reactnative/pdfviewer/PDFViewManager.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/jon/Desktop/Dev/PlantWave-NewEngine/PlantWave/React/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':android:gcanvas_library:mergeDebugNativeLibs'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > More than one file was found with OS independent path 'lib/armeabi-v7a/libfreetype.so'. If you are using jniLibs and CMake IMPORTED targets, see https://developer.android.com/studio/preview/features#automatic_packaging_of_prebuilt_dependencies_used_by_cmake

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5m 56s`

**Comment by flyskywhy** [comment]
Please use gradle-6.2 which is supported by latest RN 0.63.4 release ref to https://github.com/facebook/react-native/blob/v0.63.4/template/android/gradle/wrapper/gradle-wrapper.properties instead. When  RN upgrade to gradle-7, I will upgrade too :stuck_out_tongue_closed_eyes:

**Comment by Ayomide57** [comment]
please kindly upgrade the gradle to gradle-7, the problem still persit

**Comment by flyskywhy** [comment]
Fixed in duplicate #19
